tests/Sandpit/Exscientia/Process/test_openmm.py

```diff
@@ -1,38 +1,100 @@
-import BioSimSpace.Sandpit.Exscientia as BSS
+import math
+import re
 
+import BioSimSpace.Sandpit.Exscientia as BSS
 import pytest
 
 # Store the tutorial URL.
 url = BSS.tutorialUrl()
 
-
 @pytest.fixture(scope="session")
 def system():
     """Re-use the same molecuar system for each test."""
     return BSS.IO.readMolecules(["tests/input/ala.top", "tests/input/ala.crd"])
 
 
-def test_minimise(system):
+@pytest.mark.parametrize(
+    "restraint,dry_run",
+    [("backbone", False), ("heavy", True), ("all", True), ("none", True)]
+)
+def test_minimise(system, restraint, dry_run):
     """Test a minimisation protocol."""
 
     # Create a short minimisation protocol.
-    protocol = BSS.Protocol.Minimisation(steps=100)
+    protocol = BSS.Protocol.Minimisation(
+        steps=100, restraint=restraint, force_constant=1000
+        )
 
     # Run the process, check that it finished without error, and returns a system.
-    run_process(system, protocol)
+    new_system, process = run_process(
+        system, protocol, dry_run=dry_run
+        )
+
+    if restraint == "none":
+        return
+
+    # Check if restrained atoms stayed in place
+    restraint_atoms = system.getRestraintAtoms(restraint)
+
+    expected_lines = [
+        f"restrained_atoms = {restraint_atoms}",
+        "restraint.addBond(i, j, 0*nanometers, 418400.0 * kilojoules_per_mole / nanometer**2)"
+    ]
+
+    check_lines_in_file(expected_lines, f"{process._work_dir}/{process._name}_script.py")
```

Comment:
Thanks for all the fixing. Sorry for making so many comments.

Sorry for the confusion, when I mean checking the specific string, I mean for these integration test that calls the OpenMM to run, you just check if the atoms moves or not.

Then make a separate test like

```
@pytest.mark.parametrize("restraint", ["backbone", "heavy", "none"])
@pytest.mark.parametrize("protocol", [BSS.Protocol.Minimisation, BSS.Protocol.Equilibration, BSS.Protocol.Production])
def test_restraint_atoms(system, restraint, protocol):
    process = BSS.Process.OpenMM(system, protocol(restraint=restraint, force_constant=1000))
    with open(f"{process._work_dir}/{process._name}_script.py", 'r') as f:
        text = f.read()
    if restraint != "none":
        assert f"restrained_atoms = [1, 2, 3, 4, 5]" in text
    else:
        assert f"restrained_atoms = [1, 2, 3, 4, 5]" not in text
```
I kind of just write this to show what I mean, I cannot guarantee that it runs. In this case, you know the input system so you know what are the restrained_atoms that you test this numerically. The major advantage is that this is quick to run and shall catch any problem if something in the middle changes by other people and it is very clear about what is expected.

Then you could do something similar for `418400.0 * kilojoules_per_mole / nanometer**2` as well.

Thanks for sticking with all the comments. I think we are nearly there now. I'm happy with other part of the code. Just 
Add the test for restrained_atoms and force constant
Remove the test for the content from the `test_minimise`, `test_production` and we are ready to go.

Reply:
That's alright! Definitely makes sense to factor these two things into two different tests 👍